### PR TITLE
chore(typing): Add typing for snql fields

### PR DESCRIPTION
**Pull request by armenzg**

### CODE REVIEW

File: src/sentry/search/events/datasets/function_aliases.py [hunk 1]
```diff
@@ -11,16 +11,19 @@
...
 from sentry.search.events.types import SelectType
 from sentry.sentry_metrics.configuration import UseCaseKey
+from sentry.sentry_metrics.use_case_id_registry import UseCaseID
 
 
 def resolve_project_threshold_config(
+    # See resolve_tag_value signature
     tag_value_resolver: Callable[
-        [Optional[UseCaseKey], int, Sequence[int]], Optional[Union[int, str]]
+        [Union[UseCaseID, UseCaseKey], int, str], Optional[Union[int, str]]
```
Comment (armenzg) [inline src/sentry/search/events/datasets/function_aliases.py:20]: In order to match the definition of `resolve_tag_value`: `use_case_id: Union[UseCaseID, UseCaseKey], org_id: int, string: str `

See function: https://github.com/getsentry/sentry/blob/cf2dd1348ee1e8bf9654972705b986815306637c/src/sentry/sentry_metrics/utils.py#L151-L159

File: src/sentry/search/events/datasets/function_aliases.py [hunk 2]
```diff
@@ -11,16 +11,19 @@
...
     ],
-    column_name_resolver: Callable[[Optional[UseCaseKey], int, Sequence[int]], str],
+    # See resolve_tag_key signature
+    column_name_resolver: Callable[[Union[UseCaseID, UseCaseKey], int, str], str],
```
Comment (armenzg) [inline src/sentry/search/events/datasets/function_aliases.py:23]: In order to match `resolve_tag_key(use_case_id: Union[UseCaseID, UseCaseKey], org_id: int, string: str)` 

See definition: https://github.com/getsentry/sentry/blob/cf2dd1348ee1e8bf9654972705b986815306637c/src/sentry/sentry_metrics/utils.py#L141-L148

File: src/sentry/search/events/datasets/function_aliases.py [hunk 3]
```diff
@@ -11,16 +11,19 @@
 from sentry.search.events import constants
 from sentry.search.events.types import SelectType
 from sentry.sentry_metrics.configuration import UseCaseKey
+from sentry.sentry_metrics.use_case_id_registry import UseCaseID
 
 
 def resolve_project_threshold_config(
```
Comment (armenzg) [inline src/sentry/search/events/datasets/function_aliases.py:17]: The only caller to this function can be found here: https://github.com/getsentry/sentry/blob/cf2dd1348ee1e8bf9654972705b986815306637c/src/sentry/snuba/metrics/fields/snql.py#L735-L745

**Review by asottile-sentry — APPROVED** [review]

**Review by wmak — APPROVED** [review]

**Pull request merged**